Wikipedia/assets/index.js

```diff
@@ -703,44 +703,32 @@ function getStretchRatio(image){
     return 1.0;
 }
 
-function getDictionaryFromSrcset(srcset) {
-    /*
-     Returns dictionary with density (without "x") as keys and urls as values.
-     Parameter 'srcset' string:
-     '//image1.jpg 1.5x, //image2.jpg 2x, //image3.jpg 3x'
-     Returns dictionary:
-     {1.5: '//image1.jpg', 2: '//image2.jpg', 3: '//image3.jpg'}
-     */
-    var sets = srcset.split(',').map(function(set) {
-                                     return set.trim().split(' ');
-                                     });
-    var output = {};
-    sets.forEach(function(set) {
-                 output[set[1].replace('x', '')] = set[0];
-                 });
-    return output;
-}
-
-function useHigherResolutionImageSrcFromSrcsetIfNecessary(image) {
-    if (image.getAttribute('srcset')){
+function useHigherResolutionImageSrcIfNecessary(image) {
+    var src = image.getAttribute('src');
+    if (src){
         var stretchRatio = getStretchRatio(image);
         if (stretchRatio > maxStretchRatioAllowedBeforeRequestingHigherResolution) {
-            var srcsetDict = getDictionaryFromSrcset(image.getAttribute('srcset'));
-            /*
-            Grab the highest res url from srcset - avoids the complexity of parsing urls
-            to retrieve variants - which can get tricky - canonicals have different paths 
-            than size variants
-            */
-            var largestSrcsetDictKey = Object.keys(srcsetDict).reduce(function(a, b) {
-              return a > b ? a : b;
-            });
-
-            image.src = srcsetDict[largestSrcsetDictKey];
-
-            if(enableDebugBorders){
-                image.style.borderWidth = '10px';
-            }
-        }
+			var pathComponents = src.split("/");
+			var filename = pathComponents[pathComponents.length - 1];
+			var sizeRegex = /^[0-9]+(?=px-)/;
+			var sizeMatches = filename.match(sizeRegex);
```

Comment:
should I also check for /thumb/ in the full path in addition to matching `^[0-9]+px-` on the filename?